server/serviceadapter.js

```diff
@@ -27,13 +27,61 @@ class ServiceAdapter {
     return false;
   }
 
+  /**
+   * Determines whether a given URL points to a collection of videos.
+   * @param {string} url
+   * @returns {boolean}
+   */
+  isCollectionURL() {
+    throw `Service ${this.serviceId} does not implement method isCollectionURL`;
```

Comment:
These should throw some kind of exception instead of just a string

Reply:
The ServiceAdapter methods now throw `IncompleteServiceAdapterException`s.